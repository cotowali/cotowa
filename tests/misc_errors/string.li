var a = "a"
println("-- $undefined_variable --")
println("a = $a; b = $b; c = $c;")

println("${ a + 1 } ${ x + 1 }")
println("${ 1 + "${ a + 1 } $x"} $x")

fn string_does_not_support_index_assign() {
  "a"[0] = "a"
  var s = "abc"
  s[1] = "a"
}
