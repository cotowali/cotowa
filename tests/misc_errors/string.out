error: string.li 2,13: variable `undefined_variable` is not defined
    2 | println("-- $undefined_variable --")
      |             ^^^^^^^^^^^^^^^^^^^
error: string.li 3,22: variable `b` is not defined
    3 | println("a = $a; b = $b; c = $c;")
      |                      ^^
error: string.li 3,30: variable `c` is not defined
    3 | println("a = $a; b = $b; c = $c;")
      |                              ^^
error: string.li 5,13: mismatched types: `string` (left) and `int` (right)
    5 | println("${ a + 1 } ${ x + 1 }")
      |             ^^^^^
error: string.li 5,24: variable `x` is not defined
    5 | println("${ a + 1 } ${ x + 1 }")
      |                        ^
error: string.li 6,13: mismatched types: `int` (left) and `string` (right)
    6 | println("${ 1 + "${ a + 1 } $x"} $x")
      |             ^^^^^^^^^^^^^^^^^^^
error: string.li 6,21: mismatched types: `string` (left) and `int` (right)
    6 | println("${ 1 + "${ a + 1 } $x"} $x")
      |                     ^^^^^
error: string.li 6,29: variable `x` is not defined
    6 | println("${ 1 + "${ a + 1 } $x"} $x")
      |                             ^^
error: string.li 6,34: variable `x` is not defined
    6 | println("${ 1 + "${ a + 1 } $x"} $x")
      |                                  ^^
error: string.li 9,3: `string` does not support index assignment
    9 |   "a"[0] = "a"
      |   ^^^^^^
error: string.li 11,3: `string` does not support index assignment
   11 |   s[1] = "a"
      |   ^^^^
